src/main/scala/gw/cda/api/outputwriter/OutputWriter.scala

```diff
@@ -74,9 +171,724 @@ trait OutputWriter {
+   *
+   * @param dialect
+   * @param dbProductName
+   * @param tableName
+   * @param fieldName
+   * @param fieldDataType
+   * @param fieldNullable
+   * @return Column definition - COLUMN_NAME TYPE_DECLARATION NULLABLE (i.e. '"ColumnName" VARCHAR(1333) NOT NULL').
+   */
+  def buildDDLColumnDefinition(dialect: JdbcDialect, dbProductName: String, tableName: String, fieldName: String, fieldDataType: DataType, fieldNullable: Boolean): String = {
+    val columnDefinition = new StringBuilder()
+
+    // TODO Consider making gwcbi___seqval_hex a smaller varchar than (1333) since it is part of a clustered index
+    // Explicitly set the data type for string data to avoid nvarchar(max) and varchar2 types that are potentially too long or short.
+    // nvarchar(max) columns can't be indexed.  Oracle JDBC converts the string datatype to VARCHAR2(255) which is potentially too short.
+    val stringDataType = dbProductName match {
+      case "Microsoft SQL Server" => "VARCHAR(1333)"
+      case "PostgreSQL"           => "VARCHAR(1333)"
+      case "Oracle"               => "VARCHAR2(1333)"
+      case _                      => throw new SQLException(s"Unsupported database platform: $dbProductName")
+    }
+    // Also for string data we need to handle very large text columns that we know of to avoid truncation sql exceptions.
+    val largeStringDataType = dbProductName match {
+      case "Microsoft SQL Server" => "VARCHAR(max)"
+      case "PostgreSQL"           => "VARCHAR"
+      case "Oracle"               => "VARCHAR2(32767)" // requires MAX_STRING_SIZE Oracle parameter to be set to EXTENDED.
+      case _                      => throw new SQLException(s"Unsupported database platform: $dbProductName")
+    }
+    // Also for BLOB data we need to handle differently for different platforms.
+    val blobDataType = dbProductName match {
+      case "Microsoft SQL Server" => "VARBINARY(max)"
+      case "PostgreSQL"           => "bytea"
+      case "Oracle"               => "BLOB"
+      case _                      => throw new SQLException(s"Unsupported database platform: $dbProductName")
+    }
+    val fieldDataTypeDefinition = if (fieldDataType == StringType)
+    // TODO Consider making the determination for the need for very large text columns configurable.
+    // These are the OOTB columns we have found so far.
+      if ((tableName.equals("cc_outboundrecord") && fieldName.equals("content"))
+        || (tableName.equals("cc_contactorigvalue") && fieldName.equals("origvalue"))
+        || (tableName.equals("pc_diagratingworksheet") && fieldName.equals("diagnosticcapture"))
+        || (tableName.equals("cc_note") && fieldName.equals("body"))
+      ) largeStringDataType
+      else stringDataType
+    else if (fieldDataType == BinaryType) blobDataType
+    else getJdbcType(fieldDataType, dialect).databaseTypeDefinition
+    val nullable = if (!fieldNullable) "NOT NULL" else ""
```

Comment:
```suggestion
    val nullableQualifier = if (!fieldNullable) "NOT NULL" else ""
```
In other parts of the code, "nullable" is a boolean. I think this name expresses the intent more clearly.

Reply:
agreed. completed locally.